### PR TITLE
Carrion rebalance and update

### PR DESCRIPTION
**Pull request by TheShown911**
## About The Pull Request
This PR evolved from a carrion control spider rebalance, to a carrion mini-update with new spiders and balance changes.
This includes:

<details>
<summary>Control spider changes</summary>
Control spiders have been a very contentious topic regarding carrions, by far they were the strongs spiders and would make carrions just wordlessly kill people with little to no consequence, and in my opinion this can't be changed without removing full player control.

- Control spiders can't be used to control humanoids or mobs with a client.
- Control spiders do not carry over damage from the host mob the the owner.
- Control spider gene cost reduced from 10 to 5, and chem cost reduced from 40 to 25.
- Control spider cooldown reduced from 5 to 2 minutes.

With these changes the control spider becomes much more situational with the possibilty of combo-ing it with other spiders.
</details>

<details>
<summary>Overall carrion changes</summary>

- Carrion organs <b>cannot</b> be detected by the medical scanner anymore, so the only detection ways left are surgery or NT.
- Carrions can't eat roach flesh anymore, so they would need human flesh/organs to get evolution points.
- By using "consume the flesh" ability from the maw, carrions can rip off bodyparts from grabed humans.
- Carrions can pop spiders out of their hosts.
- With the spinneret they can make a web for 5 chemicals.
</details>

<details>
<summary>New spiders</summary>

(from left to right)
- Talking spider - allows you to speak through the mouth of the spiders host.
- Spark spider - very cheap, upon activation makes sparks and dies.
- Smooth spider - slippery, makes a scary sound when activated.
- Blight spider - injects the host with cryptobiolin, making him confused (walk in random directions, same effect as being drunk or flashbanged).
</details>

## Why It's Good For The Game
The most powerful carrion spider has been nerfed, but they are made harder to detect wich will hopefuly make them more likely to take risks. New spiders are not too powerful in direct combat (except the blight spider) but they can make fun scenarios and can be useful with proper set-up.

## Changelog
:cl: TheShown
add: Talking spider - allows you to speak through the mouth of the host.
add: Spark spider - very cheap, upon activation makes sparks and dies.
add: Smooth spider - slippery, makes a scary sound when activated.
add: Blight spider - injects the host with cryptobiolin, making him confused.
add: Consume the flesh now lets you tear off human limbs.
add: Carrions can make webs with the spinneret.
qol: Carrions get a pop-out button for their spiders.
balance: Carrion organs cannot be detected by the medbay scanner.
balance: Carrion can't eat roach meat for gene points anymore.
balance: Control spiders can't be used to control humanoids or mobs with a client.
balance: Control spiders do not carry over damage from the host mob the the owner.
balance: Control spider gene cost reduced from 10 to 5, and chem cost reduced from 40 to 25.
balance: Control spider cooldown reduced from 5 to 2 minutes.
fix: Player controled spiders and roaches now lose their AI.
/:cl:

### CODE REVIEW

**Comment by maikilangiolo** [comment]
If I can have a say in this, I think it is a great fix. Doesn't ouright remove it, allows for shenanigans but also intelligent use of the control spider without completely removing any chance of fight or retaliation from other players. Smart carrions can hide mobs in lockers and surprise attack people, with the exception that mob attacks are way more survivable that being hosed down with gamer Dallas.

Hell, carrions could attach explosive and gas spiders to roaches and have suicide bombers, or a mobile camera with observer spiders.

**Comment by MadmanMartian** [comment]
One issue I can see with this: There is NO sanity on what can be taken over.

Imagine a Carrion rocking in with the fortress boss?

**Comment by TheShown911** [comment]
That sounds like fun fun fun

**Review by MadmanMartian — CHANGES_REQUESTED** [review]
Add a mob flag so they can't be taken over, add it to:
 - Nonorganic mobs (Hivemind, drones)
 - Boss mobs/megafauna (One-star mech)
 - Mechs (They're a mob for some reason)

**Comment by TheShown911** [comment]
Converted to draft so i can give carrions more fun tools with this nerf, instead of just gutting them.

**Comment by TheShown911** [comment]
> Add a mob flag so they can't be taken over, add it to:
> 
> * Nonorganic mobs (Hivemind, drones)
> * Boss mobs/megafauna (One-star mech)
> * Mechs (They're a mob for some reason)

Updated to use the old `mob_classification (CLASSIFICATION_SYNTHETIC)` flags, and added it to more mobs.

**Comment by SirRichardFrancis** [comment]
"balance: Carrion organs cannot be detected by the medbay scanner."
Balance indeed.

File: code/game/gamemodes/changeling/modularchangling.dm [hunk 1]
```diff
@@ -61,6 +73,12 @@ var/list/datum/power/carrion/powerinstances = list()
 	genomecost = 5
 	spiderpath = /obj/item/weapon/implant/carrion_spider/mindboil
 
+/datum/power/carrion/talking_spider
+	name = "Talking spider"
+	desc = "Creates a spider that can highjack someones vocal cords, giving you the ability to talk through them."
```
Comment (MadmanMartian) [inline code/game/gamemodes/changeling/modularchangling.dm:78]:
```suggestion
	desc = "Creates a spider that can hijack someones vocal cords, giving you the ability to talk through them."
```

**Comment by TheShown911** [comment]
Please review this

**Review by Iglpigl — APPROVED** [review]

File: code/modules/mob/living/carbon/superior_animal/life.dm [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /mob/living/carbon/superior_animal/proc/check_AI_act()
-	if ((stat != CONSCIOUS) || !canmove || resting || lying || stasis || AI_inactive)
+	if ((stat != CONSCIOUS) || !canmove || resting || lying || stasis || AI_inactive || client)
 		stance = HOSTILE_STANCE_IDLE
 		target_mob = null
 		walk(src, 0)
```
Comment (Evankhell561) [inline code/modules/mob/living/carbon/superior_animal/life.dm:5]: uhm

Comment (TheShown911) [reply code/modules/mob/living/carbon/superior_animal/life.dm:5]: ??

Comment (Evankhell561) [reply code/modules/mob/living/carbon/superior_animal/life.dm:5]: this is the only change to the proc life of the mobs that I have seen. But I already tested it and there is nothing wrong. So, I don't see that it is destroying the MC.

**Pull request merged**